compiler/src/dotty/tools/dotc/util/CityHash.scala

```diff
@@ -0,0 +1,186 @@
+package dotty.tools.dotc.util
+
+/* Ported from https://github.com/google/cityhash/blob/master/src/city.cc */
```

Comment:
You should preserve the copyright/license header of the original file, and you should update AUTHORS.md to note this.